packages/core-ethereum/src/commitment.ts

```diff
@@ -4,9 +4,18 @@
 //
 // We need to persist this string of commitments in the database, and support
 // syncing back and forth with those that have been persisted on chain.
-import { iterateHash, recoverIteratedHash, HoprDB, Hash } from '@hoprnet/hopr-utils'
-import { randomBytes } from 'crypto'
-import { debug } from '@hoprnet/hopr-utils'
+import {
+  ChannelEntry,
+  debug,
+  Hash,
+  HoprDB,
+  iterateHash,
+  recoverIteratedHash,
+  toU8a,
+  u8aConcat
+} from '@hoprnet/hopr-utils'
+import { deriveCommitmentSeed } from '@hoprnet/hopr-utils/lib/crypto/commitment/keyDerivation'
+import { toUtf8Bytes } from 'ethers/lib/utils'
```

Comment:
Why don't use `new TextEncoder().encode(...)`?

Reply:
Ok, didn't know that!